docgen/extract/extract.go

```diff
@@ -0,0 +1,78 @@
+package extract
+
+import (
+	"reflect"
+
+	"github.com/golang/protobuf/proto"
+	google_protobuf "github.com/golang/protobuf/protoc-gen-go/descriptor"
+	"github.com/grpc-ecosystem/grpc-gateway/protoc-gen-swagger/options"
+
+	"github.com/gunk/gunk/docgen/pot"
+	"github.com/gunk/gunk/reflectutil"
+)
+
+// Run extract all string from swagger options and store them into pot builder.
+func Run(file *google_protobuf.FileDescriptorProto) (pot.Builder, error) {
+	pb := pot.NewBuilder()
+
+	if proto.HasExtension(file.GetOptions(), options.E_Openapiv2Swagger) {
+		ext, err := proto.GetExtension(file.GetOptions(), options.E_Openapiv2Swagger)
+		if err != nil {
+			return nil, err
+		}
+		pb.AddTranslations(toTranslate(*ext.(*options.Swagger)))
+	}
+
+	for _, s := range file.GetService() {
+		for _, m := range s.GetMethod() {
+			if proto.HasExtension(m.GetOptions(), options.E_Openapiv2Operation) {
+				ext, err := proto.GetExtension(m.GetOptions(), options.E_Openapiv2Operation)
```

Comment:
can't you just do GetExtension, and skip the iteration if the error is "not found"? otherwise these two lines are a bit repetitive.